File: data/python/tests/test_networkx_minimum_spanning_tree.py
import unittest
from function_networkx_minimum_spanning_tree import NetworkXExample

class TestNetworkXExample(unittest.TestCase):

    def test_basic_mst(self):
        calc = NetworkXExample()
        result = calc.compute_minimum_spanning_tree('A', 'B', 4, 'B', 'C', 3)
        self.assertEqual(result, 7.0)  # Expected sum of weights for MST: 4 + 3 = 7

    def test_mst_multiple_edges(self):
        calc = NetworkXExample()
        result = calc.compute_minimum_spanning_tree('A', 'B', 1, 'B', 'C', 2)
        self.assertEqual(result, 3.0)  # Expected sum of weights for MST: 1 + 2 = 3

    def test_mst_no_edges(self):
        calc = NetworkXExample()
        result = calc.compute_minimum_spanning_tree('A', 'B', 0, 'B', 'C', 0)
        self.assertEqual(result, 0.0)  # No edges, so weight sum is 0

    def test_mst_single_edge(self):
        calc = NetworkXExample()
        result = calc.compute_minimum_spanning_tree('A', 'B', 10)
        self.assertEqual(result, 10.0)  # Only one edge with weight 10

    def test_mst_disconnected_graph(self):
        calc = NetworkXExample()
        result = calc.compute_minimum_spanning_tree('A', 'B', 5, 'C', 'D', 7)
        self.assertEqual(result, 12.0)  # Expected sum of weights for MST: 5 + 7 = 12

if __name__ == '__main__':
    unittest.main()

File: data/java/tests/test_networkx_minimum_spanning_tree.java
package com.example;

import static org.junit.jupiter.api.Assertions.assertEquals;
import org.junit.jupiter.api.Test;

public class test_networkx_minimum_spanning_tree {

    @Test
    public void test_basic_mst() {
        function_networkx_minimum_spanning_tree calc = new function_networkx_minimum_spanning_tree();
        float result = calc.compute_minimum_spanning_tree("A", "B", 4.0f, "B", "C", 3.0f);
        assertEquals(7.0f, result, 0.0001);
    }

    @Test
    public void test_mst_multiple_edges() {
        function_networkx_minimum_spanning_tree calc = new function_networkx_minimum_spanning_tree();
        float result = calc.compute_minimum_spanning_tree("A", "B", 1.0f, "B", "C", 2.0f);
        assertEquals(3.0f, result, 0.0001);
    }

    @Test
    public void test_mst_no_edges() {
        function_networkx_minimum_spanning_tree calc = new function_networkx_minimum_spanning_tree();
        float result = calc.compute_minimum_spanning_tree("A", "B", 0.0f, "B", "C", 0.0f);
        assertEquals(0.0f, result, 0.0001);
    }

    @Test
    public void test_mst_single_edge() {
        function_networkx_minimum_spanning_tree calc = new function_networkx_minimum_spanning_tree();
        float result = calc.compute_minimum_spanning_tree("A", "B", 10.0f, null, null, null);
        assertEquals(10.0f, result, 0.0001);
    }

    @Test
    public void test_mst_disconnected_graph() {
        function_networkx_minimum_spanning_tree calc = new function_networkx_minimum_spanning_tree();
        float result = calc.compute_minimum_spanning_tree("A", "B", 5.0f, "C", "D", 7.0f);
        assertEquals(12.0f, result, 0.0001);
    }
}